fastlane/lib/fastlane/actions/create_keychain.rb

```diff
@@ -11,20 +11,20 @@ def self.run(params)
         escaped_password = params[:password].shellescape
 
         if params[:name]
-          escaped_name = params[:name].shellescape
-          keychain_path = "~/Library/Keychains/#{escaped_name}"
+          keychain_path = "#{ENV['HOME']}/Library/Keychains/#{params[:name]}"
```

Comment:
Are we sure `HOME` is set on all macOS machines? 

Usually we can work with `~` just fine (and did before the PR that changed the shellescaping - so this worked in the past!) - it's just that the PR introduced a bug somewhere that messed up the path expansion (I guess?).

Reply:
I'm not sure if using `ENV['HOME']` is the correct solution here 🤔 I think the proper solution would be to maybe expand the path with `File.absolute_path`??? But I wouldn't know without actually getting this running on a CI

Reply:
(I think this actually is happening later in the code, but something in the change of shellescaping logic made it work differently than before [shellescaped path is expanded differently than non shellescaped one maybe?] - we will probably have to debug the initial PR a bit more to understand what is going on exactly)

Reply:
> I'm not sure if using `ENV['HOME']` is the correct solution here 🤔 I think the proper solution would be to maybe expand the path with `File.absolute_path`??? But I wouldn't know without actually getting this running on a CI

After doing a bit of research even i agree that in some systems `HOME` might not be set correctly.
However using `File.absolute_path ("~/Library/Keychains/")` will also not resolve the issue because the given pathname starts with a "~'' it is NOT expanded, it is treated as a normal directory name as described in the documentation of Ruby Core APIs at 
[https://ruby-doc.org/core-2.6.2/File.html#absolute_path-method](url)

Modifying the code to use `Dir.home` as it will be returning the home directory of the current user which is what "~" also does and submitting a PR in sometime.

Reply:
How did this work before the PR that introduced the failure in the second to last release then, https://github.com/fastlane/fastlane/pull/14311? This only modified how the string was handled and used, but it was using `~` before as well.

Reply:
@bhuvandaruwala @janpio @joshdholtz - I apologize for the late response, I was out of town for a few days and of course the regression. Is there anything I can do?

Reply:
Don't worry - @joshdholtz rolled your changes back and did a patch release.

If you want, you can create a _new_ PR with the same changes of your old PR and try to figure out what could have caused the observed problems and fix the causes. You know your changes best, so maybe you will be able to find the culprit quickly.

Reply:
@datajockey No worries at all 😊 Stuff happens!

And @bhuvandaruwala I think what we actually want is `File.expand_path ("~/Library/Keychains/")`. This will return something like `"/Users/josh/Library/Keychains"`

I forgot that they aren't aliases of each other 🙃